### PR TITLE
BUILD: add pkg.m4 macro

### PR DESCRIPTION
**Pull request by miked-mellanox**
* for systems where it is not installed
* replaces #382

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1233/ for details.

**Comment by miked-mellanox** [comment]
@yosefe , @shamisp - please review

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/646/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
autobot build

File: config/m4/pkg.m4 [hunk 1]
```diff
@@ -0,0 +1,214 @@
+# pkg.m4 - Macros to locate and utilise pkg-config.            -*- Autoconf -*-
+# serial 1 (pkg-config-0.24)
+# 
+# Copyright © 2004 Scott James Remnant <scott@netsplit.com>.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
```
Comment (yosefe) [inline config/m4/pkg.m4]: not sure license is OK for us

File: config/m4/pkg.m4 [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# As a special exception to the GNU General Public License, if you
+# distribute this file as part of a program that contains a
+# configuration script generated by Autoconf, you may include it under
+# the same distribution terms that you use for the rest of that program.
```
Comment (shamisp) [inline config/m4/pkg.m4]: Here is an exception that allow to distribute the file under BSD license.

Comment (miked-mellanox) [inline config/m4/pkg.m4]: yep, also approved by mellanox legal dept.

**Comment by shamisp** [comment]
good !

**Comment by shamisp** [comment]
The buildbot failure seems to be unrelated:
```
----------] 1 test from signal/test_async_timer_mt
[ RUN      ] signal/test_async_timer_mt.multithread/0

command timed out: 180 seconds without output, attempting to kill
```

@yosefe - is this a known issue ?

**Comment by yosefe** [comment]
@shamisp yes, #413

**Comment by shamisp** [comment]
:+1:

**Pull request merged**